src/gfx/pal_sorting.cpp

```diff
@@ -0,0 +1,107 @@
+
+#include "gfx/pal_sorting.hpp"
+
+#include <algorithm>
+#include <png.h>
+#include <vector>
+
+#include "helpers.h"
+
+#include "gfx/convert.hpp"
+#include "gfx/main.hpp"
+
+using std::swap;
+
+namespace sorting {
+
+void indexed(std::vector<Palette> &palettes, int palSize, png_color const *palRGB,
+             png_byte *palAlpha) {
+	options.verbosePrint(Options::VERB_LOG_ACT, "Sorting palettes using embedded palette...\n");
+
+	auto pngToRgb = [&palRGB, &palAlpha](int index) {
+		auto const &c = palRGB[index];
+		return Rgba(c.red, c.green, c.blue, palAlpha ? palAlpha[index] : 0xFF);
+	};
+
+	// HACK: for compatibility with old versions, add unused colors if:
+	// - there is only one palette, and
+	// - only some of the first N colors are being used
+	if (palettes.size() == 1) {
+		Palette &palette = palettes[0];
+		// Build our candidate array of colors
+		decltype(palette.colors) colors{UINT16_MAX, UINT16_MAX, UINT16_MAX, UINT16_MAX};
+		for (int i = 0; i < options.maxOpaqueColors(); ++i) {
+			colors[i + options.hasTransparentPixels] = pngToRgb(i).cgbColor();
+		}
+
+		// Check that the palette only uses those colors
+		if (std::all_of(palette.begin(), palette.end(), [&colors](uint16_t color) {
+			    return std::find(colors.begin(), colors.end(), color) != colors.end();
+		    })) {
```

Comment:
Is there some clang-format setting that's sometimes using four spaces for indentation? (Sorry for so many nit-picky comments about formatting, but this is the first usage of clang-format in rgbds, and I'm confused by some of its choices.)

Reply:
I reckon it's got something to do with alignment.

Reply:
Or, scratch that; I think it's about being inside of the `if`, instead?